Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,385 @@
+     *
+     * It requires to subscribe previously to a specific queue
+     *
+     * @throws \Exception
+     *
+     * @return \Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface|null
+     */
+    public function receiveNoWait()
+    {
+        if ($this->currentEnvelope) {
+            throw new \RuntimeException(
+                'AmqpQueueDriver: This driver has a message that was not acknowledged yet. A message must be processed and acknowledged before receiving new messages.'
+            );
+        }
+
+        $this->dequeueingTimeMs = 0;
+
+        $this->currentEnvelope = $this->queue->get();
+
+        $msg = null;
+
+        if (!$this->currentEnvelope) {
+            return null;
+        }
+
+        $start = microtime(true);
+        $deserializationContext = new DeserializationContext();
+
+        try{
+            /** @var QueueMessageInterface $msg */
+            $msg = $this->getSerializer()->deserialize($this->currentEnvelope->getBody(), SerializableInterface::class, $this->format, $deserializationContext);
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $this->currentEnvelope->getHeaders(), 'body' => $this->currentEnvelope->getBody()]);
+            $this->ack();
+            return null;
+        }
+        foreach ($this->currentEnvelope->getHeaders() as $header => $value) {
+            $msg->setHeader($header, $value);
+        }
+
+        // Calculate how long it took to deserilize the message
+        $this->dequeueingTimeMs = (int) ((microtime(true) - $start) * 1000);
+
+        return $msg;
+    }
+
+    /**
+     * @return int The time it took in ms to de-queue and deserialize the message
+     */
+    public function getDequeueingTimeMs()
+    {
+        return $this->dequeueingTimeMs;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge(string $queue)
+    {
+        $this->manager->getQueue($queue)->purge();
+    }
+
+    /**
+     * Sleep the script for a given number of seconds.
+     *
+     * @param int|float $seconds
+     */
+    public function sleep($seconds)
+    {
+        if ($seconds < 1) {
+            usleep((int) $seconds * 1000000);
+        } else {
+            sleep((int) $seconds);
+        }
+    }
+
+    /**
+     *
+     */
+    private function fillByContainer()
+    {
+        $this->container->get();
+    }
+
+    /**
+     * @return PhpAmqpLibQueueManager|QueueManager
+     */
+    public function getManager()
+    {
+        return $this->manager;
+    }
+
+    /**
+     * @param string $queueName
+     * @param int $flag
+     * @param array $options
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function setQueue($queueName)
+    {
+        $this->manager->declareQueue($queueName);
+    }
+
+    /**
+     * @param $exchange
+     */
+    public function setExchange($exchange)
```

Comment:
I don't see any use of this method, where is it used?

Reply:
This class was removed.